apps/dot-voting/app/store/votes.js

```diff
@@ -226,8 +229,69 @@ const marshallVote = ({
     totalParticipation: totalParticipation,
     metadata,
     executionScript,
+    executionTargetData,
     executed,
     participationPct:
       totalVoters === 0 ? 0 : (totalParticipation / totalVoters) * 100,
   }
 }
+
+const getVoteExecutionTargets = (vote) => {
+  return vote.executionTargets || []
+}
+
+const decorateVote = async (vote) => {
+  const executionTargets = getVoteExecutionTargets(vote)
+  const currentApp = await app.currentApp().toPromise()
+  if (!executionTargets.length) {
+    // If there's no execution target, consider it targetting this Dot Voting app
+    return {
+      ...vote,
+      executionTargetData: {
+        ...currentApp,
+        identifier: undefined,
+      }
+    }
+  } else if (executionTargets.length > 1) {
+    // If there's multiple targets, make a "multiple" version
+    return {
+      ...vote,
+      executionTargetData: {
+        address: '0x0000000000000000000000000000000000000000',
+        name: 'Multiple',
+        iconSrc: currentApp.icon(24),
+        identifier: undefined,
+      }
+    }
+  } else {
+    // Otherwise, try to find the target from the installed apps
+    const [targetAddress] = executionTargets
+    return app.installedApps().pipe(
+      first(),
+      mergeMap(installedApps => from(installedApps)),
+      first(
+        app => app.appAddress === targetAddress,
+        {
+          appAddress: targetAddress,
+          name: 'External',
+          icon: currentApp.icon,
```

Comment:
We might want to come up with a more informative icon for potential external forwarded actions than just the Dot Voting icon at some point, such as a question mark or something else, but there's not clear use case for this yet